From 3989e169fc0da9c29da8dd692427d4f4c1ace413 Mon Sep 17 00:00:00 2001
From: Daniel Xu <dlxu@fb.com>
Date: Wed, 23 Jun 2021 18:17:26 -0700
Subject: [PATCH] Fix ODR violation in tests

Summary: ContinuePlugin was defined twice. Namespace the test one to avoid ODR issues.

Reviewed By: davide125

Differential Revision: D29346385

fbshipit-source-id: 58fdde38a64c8bd22d14d55d0ba04ae865d2fabf
---
 src/oomd/config/ConfigCompilerTest.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/oomd/config/ConfigCompilerTest.cpp b/src/oomd/config/ConfigCompilerTest.cpp
index ecf2b92..abe5d10 100644
--- a/src/oomd/config/ConfigCompilerTest.cpp
+++ b/src/oomd/config/ConfigCompilerTest.cpp
@@ -36,6 +36,7 @@ using namespace Oomd::Config2;
 using namespace Oomd::Engine;
 
 namespace {
+
 int prerun_count;
 int prerun_stored_count;
 int count;
@@ -57,6 +58,7 @@ void reset_counters() {
 static constexpr auto kRandomCgroupFs = "oomd/fixtures/cgroup";
 
 namespace Oomd {
+namespace test {
 
 class ContinuePlugin : public BasePlugin {
  public:
@@ -308,6 +310,7 @@ REGISTER_PLUGIN(NoInit, NoInitPlugin::create);
 REGISTER_PREKILL_HOOK(NoOpPrekillHook, NoOpPrekillHook::create);
 REGISTER_PLUGIN(Kill, KillPlugin::create);
 
+} // namespace test
 } // namespace Oomd
 
 class CompilerTest : public ::testing::Test {
